docs/source/usage.rst

```diff
@@ -0,0 +1,164 @@
+  ``environment.yaml`` file, `you should do so with the
+  *pip:* key <https://conda.io/docs/user-guide/tasks/manage-environments.html#creating-an-environment-file-manually>`_.
+
+  .. note::
+
+     For a list of sample repositories, see :ref:`samples`.
+
+Supported configuration files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Below is a list of supported configuration files.
+
+``requirements.txt``
+^^^^^^^^^^^^^^^^^^^^
+
+This specifies a list of python packages that would be installed in a virtualenv (or conda environment).
+
+``environment.yml``
+^^^^^^^^^^^^^^^^^^^
+
+This is a conda environment specification, that lets you install packages with conda.
+
+.. note::
+
+   You must leave the name of the environment empty for this to work out of the box.
+
+``apt.txt``
+^^^^^^^^^^^
+
+A list of debian packages that should be installed. The base image used is usually the latest released
+version of Ubuntu (currently Zesty.)
+
+``postBuild``
+^^^^^^^^^^^^^
+
+A script that can contain arbitrary commands to be run after the whole repository has been built. If you
+want this to be a shell script, make sure the first line is `#!/bin/bash`.
+
+.. note::
+
+   This file must be executable to be used with ``repo2docker``. To do this,
+   run the following::
+
+     chmod +x postBuild
+
+``REQUIRE``
+^^^^^^^^^^^
+
+This specifies a list of Julia packages! Currently only version 0.6 of Julia is supported, but more will
+be as they are released.
+
+.. note::
+
+   Using a ``REQUIRE`` file also requires that the repository contain an
+   ``environment.yaml`` file.
+
+``Dockerfile``
+^^^^^^^^^^^^^^
+
+This will be treated as a regular Dockerfile and a regular Docker build will be performed. The presence
+of a Dockerfile will cause all other building behavior to not be triggered.
+
+Using ``repo2docker`` with a JupyterHub
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is possible to use ``repo2docker`` in order to build JupyterHub-ready
+Docker images. In order for this to work properly, **the version of the ``jupyterhub``
+package in your git repository must match the version in your JupyterHub
+deployment**. For example, if your JupyterHub deployment runs ``jupyterhub==0.8``,
+you should put the following in ``requirements.txt`` or ``environment.yaml``::
+
+  jupyterhub==0.8.*
+
+Running ``repo2docker`` locally
+-------------------------------
+
+For information on installing ``repo2docker``, see :ref:`install`.
+
+.. note::
+
+   Docker must be running on your machine in order to build images
+   with ``repo2docker``.
+
+Building an image
+~~~~~~~~~~~~~~~~~
+
+The simplest invocation of ``repo2docker`` builds a Docker image
+from a git repo, then runs a Jupyter server within the image
+so you can explore the repository's contents.
+You can do this with the following command::
+
+  jupyter-repo2docker https://github.com/jakevdp/PythonDataScienceHandbook
+
+After building (it might take a while!), it should output in your terminal
```

Comment:
remove !

Reply:
output a message in your terminal

Reply:
not sure what you'd like me to remove here